admin/menu/class-submenu-hider.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * @package WPSEO\Admin\Menu
+ */
+
+/**
+ * Hides submenu items if the advanced settings are not enabled.
+ */
+class WPSEO_Submenu_Hider implements WPSEO_WordPress_Integration {
+
+	/**
+	 * Registers all hooks to WordPress
+	 */
+	public function register_hooks() {
+		add_filter( 'wpseo_submenu_pages', array( $this, 'filter_settings_pages' ) );
+	}
+
+	/**
+	 * Filters all advanced settings pages from the given pages.
+	 *
+	 * @param array $pages The pages to filter.
+	 *
+	 * @return array Filtered list of submenu pages to show.
+	 */
+	public function filter_settings_pages( array $pages ) {
+		$options = WPSEO_Options::get_options( array( 'wpseo' ) );
+
+		if ( wpseo_advanced_settings_enabled( $options ) ) {
+			return $pages;
+		}
+
+		$pages_to_hide = WPSEO_Advanced_Settings::get_advanced_pages();
+		$page          = filter_input( INPUT_GET, 'page' );
+
+		if ( WPSEO_Advanced_Settings::is_advanced_settings_page( $page ) ) {
```

Comment:
Please add a comment about what's happening here; it's not clear from just the code alone.